### DISCUSSION TITLE
stm32f411re nucleo board communication with raspberry pi 4b.

### DISCUSSION THREAD

**Discussion by Sreejasailu** [grblHAL/core#623] [Q&A]
i have successfully imported the zip file onto the stm32cubeide. and i want to do the communication between stm32f4xx and raspberry pi4b via uart-usb.  what are the changes to be made.?
How do the communication work.?
i have done the raspberry pi 4 part, and i only need to communicate through the stm32f411re nucleo board. i want two way communication. could you please help with this. 
and i want to say that y i didn't upload the bin file, but trying to do this is, i want to send random data both ways with transmit and receive functions in it. and to be precise transmit will be integer value, receive will be a string value.

**Comment by terjeio** [comment]
> what are the changes to be made.?

None, other than creating the program running on the Pi.

> How do the communication work.?

The protocol is documented [here](https://github.com/gnea/grbl/wiki/Grbl-v1.1-Interface), grblHAL extensions [here](https://github.com/grblHAL/core/wiki/Report-extensions).

>  and to be precise transmit will be integer value, receive will be a string value.

Then you have to implement your own protocol. The [HPGL plugin](https://github.com/grblHAL/Templates/tree/master/my_plugin/hpgl) is the only example I have that completely switches out the Grbl protocol.

**Comment by Sreejasailu** [comment]
do you have any user manual or information for the basics or a newbie?
if yes, can you provide that document?

**Reply by terjeio** [reply]
> do you have any user manual or information for the basics or a newbie?

Not much more than I have already provided. [This wiki page](https://github.com/grblHAL/core/wiki/For-sender-developers) and a basic [technical reference](http://svn.io-engineering.com/grblHAL/html/) for the core is in the works but those are aimed at somewhat experienced programmers.
